proxy/envoy/discovery.go

```diff
@@ -416,7 +438,13 @@ func (ds *DiscoveryService) ListClusters(request *restful.Request, response *res
 			return
 		}
 
-		clusters := buildClusters(ds.Environment, role)
+		clusters, err := buildClusters(ds.Environment, role)
+		if err != nil {
+			// If client experiences an error, 503 error will tell envoy to keep its current
+			// cache and try again later
+			errorResponse(response, http.StatusServiceUnavailable, "CDS "+err.Error())
```

Comment:
have you verified this behavior?

Reply:
asked the question on the envoy-users google group 
https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/envoy-users/s9ka4-au9LY/QzS4ZSyHIQAJ